manifests/server/service.pp

```diff
@@ -20,7 +20,9 @@
     name     => $mysql::server::service_name,
     enable   => $mysql::server::real_service_enabled,
     provider => $mysql::server::service_provider,
-    require => [ File['mysql-config-file'], Package['mysql-server'] ]
+    # if you manage the config file outside of the module or want mysql provided by another package, you may still use this service
+    # provided you set the parameters config_file resp. package_name on the mysql::server class
+    require  => [ File[$mysql::server::config_file], Package[$mysql::server::package_name]]
```

Comment:
Why should we? Pls see the discussion on this at #630? In the proposed form it works well with a externally manged config files. I think this is the better alternative because MySQL still requires a config file to start.
This form of the require ensures that the config exists, when the server starts but even allows not standard locations of the file.
Because of this, this PR is an alternative to #630.